### ISSUE TITLE
Setting AWS credentials via config file leads to `MissingRegion` error

### ISSUE THREAD

**Issue by kvch** [elastic/beats#25263]
For confirmed bugs, please report:
- Version: 542bef8611cd24bfdeee6e6c1c9c7fe0158a3387
- Operating System: Linux
- Steps to Reproduce: 

1. Run `functionbeat deploy beatbox` with the configuration below:

```yaml
functionbeat.provider.aws.endpoint: "s3.eu-west-3.amazonaws.com"
functionbeat.provider.aws.access_key_id: "<id>"
functionbeat.provider.aws.secret_access_key: "<snipped_for_paste>"
functionbeat.provider.aws.deploy_bucket: beatbox
functionbeat.provider.aws.functions:
  - name: beatbox
    enabled: true
    type: cloudwatch_logs
    description: "lambda function for cloudwatch logs"
    triggers:
      - log_group_name: /aws/lambda/my-lambda-function
```

The configuration is correct. When setting the same options via environment variables the deployment is successful:

```sh
export AWS_ACCESS_KEY_ID=ABCDEFGHIJKLMNOPUSER
export AWS_SECRET_ACCESS_KEY=EXAMPLE567890devgHIJKMLOPNQRSTUVZ1234KEY
export AWS_DEFAULT_REGION=us-east-1
```

Error message:
```
MissingRegion: could not find region configuration
```

A similar error is reported for Filebeat s3 input. It uses the same AWS credential package as Functionbeat, so the problem is probably common between the two Beats.

**Comment by kvch** [comment]
CC @kaiyan-sheng

**Comment by kaiyan-sheng** [comment]
Hi @kvch, I think the problem is I removed setting default region under `x-pack/libbeat` in `GetAWSCredentials` function so some initial AWS API calls are failing with error `MissingRegion`. I added the default region back into libbeat code: https://github.com/elastic/beats/pull/25446. This way Functionbeat shouldn't need to rely on the `AWS_DEFAULT_REGION` environment variable.

**Comment by kaiyan-sheng** [comment]
Closing this issue because it's fixed by #25446. Please feel free to reopen or create a new issue if this didn't fix the problem. Thank you!!
